src/DI/ElasticsearchExtension.php

```diff
@@ -23,29 +23,32 @@ public function getConfigSchema(): Schema
 				Expect::structure([
 					'host' => Expect::string()->required(),
 					'port' => Expect::int(),
-					'schema' => Expect::string(),
+					'scheme' => Expect::string(),
 					'path' => Expect::string(),
 					'user' => Expect::string(),
 					'pass' => Expect::string(),
 				])->castTo('array')
 			)),
-			'retries' => Expect::int(),
+			'retries' => Expect::int(1),
 		]);
 	}
 
 	public function beforeCompile(): void
 	{
 		$config = $this->config;
+		$config->hosts = $this->getHosts();
 		$builder = $this->getContainerBuilder();
 
-		$builder->addDefinition($this->prefix('clientBuilder'))
-			->setType(ClientBuilder::class)
-			->setFactory([ClientBuilder::class, 'create'])
-			->setArguments($config->hosts);
-
 		$builder->addDefinition($this->prefix('client'))
 			->setType(Client::class)
-			->setFactory(['@' . $this->prefix('clientBuilder'), 'build']);
+			->setFactory([ClientBuilder::class, 'fromConfig'])
+			->setArguments([(array) $config]);
+	}
+
+
+	private function getHosts(): array
+	{
+		return count($this->config->hosts) > 0 ? $this->config->hosts : ['localhost:9200'];
```

Comment:
Better do nothing if hosts are empty, or set hosts to require at least one item

Reply:
@mabar No problem, updated.